src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5737,19 +5807,24 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
                     snapshotBuilder = null;
                 }
 
+                LocalFunctionSymbol updatedLocalFunction = node.Symbol.WithContainingSymbol(GetUpdatedContainingSymbol(node.Symbol));
+                _reinferredVariablesOpt?.SetReinferredSymbol(node.Symbol, updatedLocalFunction);
+
                 Analyze(compilation,
-                        node.Symbol,
+                        updatedLocalFunction,
                         body,
                         _binder,
                         _conversions,
                         Diagnostics,
-                        useMethodSignatureParameterTypes: false,
-                        methodSignatureOpt: null,
+                        lambdaSymbolOpt: null,
                         initialState: GetVariableState(this.TopState()),
                         analyzedNullabilityMap,
                         updatedSymbolMap,
+                        reinferredVariablesOpt: _reinferredVariablesOpt?.EnterNewFunction(),
                         snapshotBuilder,
                         returnTypesOpt: null);
+
+                SetUpdatedSymbol(node, node.Symbol, updatedLocalFunction);
```

Comment:
>SetUpdatedSymbol(node, node.Symbol, updatedLocalFunction); [](start = 16, length = 58)

It feels like we would want to re-map references to type parameters of remapped local functions.

Reply:
[https://github.com/dotnet/roslyn/issues/38922](https://github.com/dotnet/roslyn/issues/38922) to track.

---
In reply to: [328236924](https://github.com/dotnet/roslyn/pull/38769#discussion_r328236924) [](ancestors = 328236924)